tests/sample_qc/test_de_novo.py

```diff
@@ -0,0 +1,230 @@
+            assert result == pytest.approx(
+                expected, rel=1e-6
+            )  # Compare floating point values safely
+
+    @pytest.mark.parametrize(
+        "pl_input, expected",
+        [
+            (
+                [0, 10, 100],
+                [0.9090909090082644, 0.09090909090082644, 9.090909090082645e-11],
+            ),
+            ([0, 0, 0], [0.3333333333333333, 0.3333333333333333, 0.3333333333333333]),
+        ],
+    )
+    def test_transform_pl_to_pp(self, pl_input, expected) -> None:
+        """
+        Test PL to PP transformation.
+
+        :param pl_input: Input PL values.
+        :param expected: Expected PP values.
+        :return: None.
+        """
+        expr = transform_pl_to_pp(hl.literal(pl_input))
+        result = hl.eval(expr)
+
+        assert result == pytest.approx(
+            expected, abs=1e-12
+        ), f"Got {result}, expected {expected}"
+
+    @pytest.mark.parametrize(
+        "locus_key, is_xx, expected_diploid, expected_hemi_x, expected_hemi_y",
+        [
+            ("autosomal", True, True, False, False),
+            ("autosomal", False, True, False, False),
+            ("par1", True, True, False, False),
+            ("par2", False, True, False, False),
+            ("x_nonpar", True, True, False, False),
+            ("x_nonpar", False, False, True, False),
+            ("y_nonpar", True, False, False, False),
+            ("y_nonpar", False, False, False, True),
+        ],
+    )
+    def test_get_copy_state_by_sex(
+        self, locus_key, is_xx, expected_diploid, expected_hemi_x, expected_hemi_y
+    ) -> None:
+        """
+        Test copy state determination based on locus type and sex.
+
+        :param locus_key: Locus key.
+        :param is_xx: Whether the individual is XX.
+        :param expected_diploid: Expected diploid state.
+        :param expected_hemi_x: Expected hemizygous X state.
+        :param expected_hemi_y: Expected hemizygous Y state.
+        :return: None.
+        """
+        locus = self.loci[locus_key]
+        is_xx_expr = hl.literal(is_xx)
+
+        diploid, hemi_x, hemi_y = get_copy_state_by_sex(locus, is_xx_expr)
+        result = hl.eval([diploid, hemi_x, hemi_y])
+
+        assert result == [
+            expected_diploid,
+            expected_hemi_x,
+            expected_hemi_y,
+        ], (
+            f"Failed for locus={locus}, is_xx={is_xx}. Expected"
+            f" {[expected_diploid, expected_hemi_x, expected_hemi_y]}, got {result}"
+        )
+
+    @pytest.mark.parametrize(
+        "locus_key, proband_gt, father_gt, mother_gt, is_xx, expected",
+        [
+            ("autosomal", (0, 1), (0, 0), (0, 0), False, True),
+            ("autosomal", (1, 1), (0, 0), (0, 0), False, False),
+            ("x_nonpar", (1, 1), None, (0, 0), False, True),
+            ("x_nonpar", (1, 1), (0, 0), None, False, None),
+            ("y_nonpar", (1, 1), None, None, False, None),
```

Comment:
these are testing for `None` as a return value when the function docstring specifies that it only returns boolean expressions. can you add to the docstring that it will return None if there is a missing genotype?